common/slow_log_timer.h

```diff
@@ -0,0 +1,75 @@
+/// Copyright 2016 Pinterest Inc.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+/// http://www.apache.org/licenses/LICENSE-2.0
+
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+
+//
+// @author shu (shu@pinterest.com)
+//
+
+#pragma once
+
+#include <glog/logging.h>
+#include <sys/time.h>
+#include <stdlib.h>
+#include <string>
+
+#include "common/stats/stats.h"
+#include "common/timer.h"
+
+namespace common {
+
+// A timer which uses GLOG to log slow requests (long Timer object life cycle)
+// and logs the content with a life cycle time threshold and a sample rate.
+class SlowLogTimer : public common::Timer {
+ public:
+  explicit SlowLogTimer(const uint32_t metric,
+                        const std::string& log_message,
```

Comment:
change it to "std::string log_message" and use "std::move()" at line 39 so that we could potentially use move semantic.